jerry-core/CMakeLists.txt

```diff
@@ -280,6 +298,88 @@ set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_GLOBAL_HEAP_SIZE=${JERRY_GLOBAL_HEAP_SI
 # Maximum size of stack memory usage
 set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_STACK_LIMIT=${JERRY_STACK_LIMIT})
 
+## Read "jerryscript-config.h.in" for default values
+function(read_set_defines FILE PREFIX OUTPUTVAR)
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/${FILE}" INPUT_FILE_CONTENTS)
+
+  # match all "#define <PREFIX>\n" lines
+  # notes:
+  #  * before the "#" there must be a newline and any number of spaces.
+  #  * after the "#" there can be any number of spaces.
+  string(REGEX MATCHALL "\r?\n[ ]*#[ ]*define ${PREFIX}[^\n]*"
+         RAW_DEFINES "${INPUT_FILE_CONTENTS}")
+
+  set(SELECTED_VARS )
+
+  # Transform the defines to a list of (<name>; <value>; <name 2>; <value 2>; ...) list
+  foreach(DEFINE_ENTRY ${RAW_DEFINES})
+    # by default every define value is empty
+    set(DEFINE_VALUE " ")
+
+    # split up the define at the space between the define name and value (if there is any)
+
+    # first remove "#define" part of the string
+    string(REGEX REPLACE "\r?\n[ ]*#[ ]*define[ ]+" "" DEFINE_KEY_VALUE "${DEFINE_ENTRY}")
+    string(FIND "${DEFINE_KEY_VALUE}" " " DEFINE_KEY_IDX)
+    string(LENGTH "${DEFINE_KEY_VALUE}" DEFINE_LENGTH)
+
+    if (DEFINE_KEY_IDX EQUAL "-1")
+        set(DEFINE_KEY ${DEFINE_KEY_VALUE})
+    else()
+        string(SUBSTRING "${DEFINE_KEY_VALUE}" 0 ${DEFINE_KEY_IDX} DEFINE_KEY)
+        string(SUBSTRING "${DEFINE_KEY_VALUE}" ${DEFINE_KEY_IDX} 64 DEFINE_VALUE)
```

Comment:
Why is the substring length 64? We could use -1 if we want the remaining string.

Reply:
ohh.. cool! thanks that works